src/core/metrics/app-metrics.ts

```diff
@@ -136,29 +153,43 @@ export class AppMetrics {
     this.processEvent(e.detail.name, e.detail.payload);
   }
 
-  watchList = METRICS_WATCH_LIST;
+  watch(name: string, payload?: unknown) {
+    // if current_event not available do not wait analytics etc
+    // TODO: use runtimeDepsTree after further analysis
+    if (name === AppFeature.CURRENT_EVENT && payload === false) {
+      delete this.watchList[AppFeature.ANALYTICS_PANEL];
+      delete this.watchList['_done_layersInAreaAndEventLayerResource'];
+    }
 
-  watch(name: string) {
     if (this.watchList[name] === null) {
       const timing = performance.now();
       this.watchList[name] = timing;
+
       if (every(this.watchList, Boolean)) {
-        // we need to wait EVENT_MAP_IDLE after all events from watchList
+        // watchList completed
+
+        // we need to wait (again) EVENT_MAP_IDLE after all events from watchList
         // to determite that app is ready
         if (name !== EVENT_MAP_IDLE) {
+          if (KONTUR_METRICS_DEBUG) {
+            console.warn('metrics waiting final mapIdle', this.watchList);
+          }
+
           this.watchList[EVENT_MAP_IDLE] = null;
-          return;
-        }
 
-        setTimeout(() => {
+          // repaint to ensure EVENT_MAP_IDLE will be triggered
+          // FIXME: replace globalThis?.KONTUR_MAP after init refactor
+          globalThis?.KONTUR_MAP.triggerRepaint();
+        } else {
+          // app is ready
+          this.cleanup();
           this.report('ready', timing);
-          const eventReadyEvent = new Event('event_ready_for_screenshot');
-          window.dispatchEvent(eventReadyEvent);
-        }, 299); // extra time to prevent rendering glitches
+          this.sendReports();
 
-        // watchList done
-        this.cleanup();
-        return;
+          // if timing > 9000 it's too late
+          readyForScreenshot(globalThis?.KONTUR_MAP);
```

Comment:
```suggestion
          emitReadyForScreenshot(globalThis?.KONTUR_MAP);
```
or something?
it's just `readyForScreenshot` is a name for boolean or string variable